src/flatland/ordered/map.clj

```diff
@@ -86,7 +86,8 @@
     (.equiv this other))
   (hashCode [this]
     (reduce (fn [acc ^MapEntry e]
-              (let [k (.key e), v (.val e)]
+              (let [k (if-some [k (.key e)] k 0)
```

Comment:
We shouldn't be substituting a 0 and then hashing that 0 - we should use the 0 instead of hashing at all, to remain compatible with `clojure.lang.APersistentMap/hashMap`. Your version works because `0` happens to hash to `0`, but we shouldn't rely on that non-guarantee.